spec/checkMergeConflictSpec.js

```diff
@@ -0,0 +1,216 @@
+require('dotenv').config();
+const { createProbot } = require('probot');
+// The plugin refers to the actual app in index.js.
+const oppiaBot = require('../index');
+const checkMergeConflictModule = require('../lib/checkMergeConflicts');
+const scheduler = require('../lib/scheduler');
+let payload = require('../fixtures/pullRequestPayload.json');
+
+describe('Merge Conflict Check', () => {
+  /**
+   * @type {import('probot').Probot} robot
+   */
+  let robot;
+
+  /**
+   * @type {import('probot').Octokit} github
+   */
+  let github;
+
+  /**
+   * @type {import('probot').Application} app
+   */
+  let app;
+
+  let mergeConflictLabel = {
+    id: 971968901,
+    node_id: 'MDU6TGFiZWw5NzE5Njg5MDE=',
+    url:
+      "https://api.github.com/repos/oppia/oppia/labels/PR:%20don't%20merge%20-%20HAS%20MERGE%20CONFLICTS",
+    name: "PR: don't merge - HAS MERGE CONFLICTS",
+    color: 'd93f0b',
+    default: false,
+  };
+  beforeEach(() => {
+    spyOn(scheduler, 'createScheduler').and.callFake(() => {});
+
+    github = {
+      issues: {
+        addLabels: jasmine.createSpy('addLabels').and.resolveTo({}),
+        addAssignees: jasmine.createSpy('addAssignees').and.resolveTo({}),
+        createComment: jasmine.createSpy('createComment').and.resolveTo({}),
+        removeLabel: jasmine.createSpy('removeLabel').and.resolveTo({}),
+      },
+    };
+
+    robot = createProbot({
+      id: 1,
+      cert: 'test',
+      githubToken: 'test',
+    });
+
+    app = robot.load(oppiaBot);
+    spyOn(app, 'auth').and.resolveTo(github);
+    spyOn(
+      checkMergeConflictModule,
+      'checkMergeConflictsInAllPullRequests'
+    ).and.callThrough();
+    spyOn(
+      checkMergeConflictModule,
+      'checkMergeConflictsInPullRequest'
+    ).and.callThrough();
+  });
+
+  describe('when pull request gets merged', () => {
+    beforeEach(async () => {
+      // Simulate a merged conflict in pull request.
+      payload.payload.pull_request.merged = true;
+      payload.payload.action = 'closed';
+
+      const pullRequest = { ...payload.payload.pull_request };
+      pullRequest.merged = false;
+      pullRequest.mergeable = false;
+
+      github.pulls = {
+        get: jasmine.createSpy('get').and.resolveTo({
+          data: pullRequest,
+        }),
+        list: jasmine.createSpy('list').and.resolveTo({
+          data: [pullRequest],
+        }),
+      };
+
+      await robot.receive(payload);
+    });
+
+    it('should check for merge conflict', () => {
+      expect(
+        checkMergeConflictModule.checkMergeConflictsInAllPullRequests
+      ).toHaveBeenCalled();
+      expect(
+        checkMergeConflictModule.checkMergeConflictsInPullRequest
+      ).toHaveBeenCalled();
+    });
+
+    it('ping pr author regarding merge conflict', () => {
+      expect(github.issues.createComment).toHaveBeenCalled();
+
+      const link = 'link'.link(
+        'https://help.github.com/articles/resolving-a-merge-conflict-using-the-command-line/'
```

Comment:
Is this line too long? Please break if yes.

Reply:
Done.